components/places/android/src/main/java/mozilla/appservices/places/PlacesConnection.kt

```diff
@@ -1079,3 +1139,81 @@ data class VisitInfosWithBound(
         }
     }
 }
+
+/**
+ * A helper extension method for conveniently measuring execution time of a closure.
+ *
+ * N.B. since we're measuring calls to Rust code here, the provided callback may be doing
+ * unsafe things. It's very imporant that we always call the function exactly once here
+ * and don't try to do anything tricky like stashing it for later or calling it multiple times.
+ */
+inline fun <U> TimingDistributionMetricType.measure(funcToMeasure: () -> U): U {
+    val timerId = this.start()
+    try {
+        return funcToMeasure()
+    } finally {
```

Comment:
I agree, but also, it's complicated. We want to timer as close to the underlying DB call as possible so that we're not measuring framework overhead stuff, but it's awkward for code at that level to tell whether the containing function call is going to succeed for fail. I think it will be cleaner when we move this into the rust layer, which can catch and ignore db-related errors without worrying about any of the others.

(Relatedly, it probably is worth trying to measure framework overhead somehow, because consumers will certainly care about it. But I think we're too *low* in the stack here to measure that successfully).